src/recon_buildblock/KappaComputation.cxx

```diff
@@ -0,0 +1,264 @@
+{
+  use_approximate_hessian = use_approximate;
+}
+
+template <typename TargetT>
+bool
+KappaComputation<TargetT>::
+get_compute_with_penalty()
+{
+  return compute_with_penalty;
+}
+
+template <typename TargetT>
+void
+KappaComputation<TargetT>::
+set_compute_with_penalty(bool with_penalty)
+{
+  compute_with_penalty = with_penalty;
+}
+
+template <typename TargetT>
+void
+KappaComputation<TargetT>::
+process_data()
+{
+  input_image = read_from_file<TargetT>(input_image_filename);
+  // Unfortunately we have to setup. Therefore we will cheat with regards to sensitivities that are unused in the kappa
+  // computation.
+  // To prevent sensitivity from being computed (which takes a lot of computation), the sensitivity filename is set to
+  // a filename known to exist, one of the file inputs, and re-computation is turned off.
+  objective_function_sptr->set_recompute_sensitivity(false);
+  objective_function_sptr->set_use_subset_sensitivities(false);
+  objective_function_sptr->set_sensitivity_filename(input_image_filename);
+  objective_function_sptr->set_up(input_image);
+
+  if (get_use_approximate_hessian())
+  {
+    // Compute the kappa image using the approximate hessian
+    // The input image is used as a template for this method
+    compute_kappa_with_approximate();
+  }
+  else
+  {
+    //Compute the kappa image with the full hessian at the current image estimate.
+    // The input image here is assumed to be the current_image_estimate at the point the hessian will be computed
+    compute_kappa_at_current_image_estimate();
+  }
+
+  // Save the output
+  output_file_format_sptr->write_to_file(kappa_filename, *kappa_image_target_sptr);
+
+  info("Spatially variant penalty strength (Kappa) has been computed and saved as " + kappa_filename + ".");
+}
+
+template <typename TargetT>
+void
+KappaComputation<TargetT>::
+compute_kappa_at_current_image_estimate()
+{
+  reset_kappa_image_target_sptr(input_image);
+  info("Computing the spatially variant penalty strength at the current image estimate, this may take a while.");
+  auto ones_image_sptr = input_image->get_empty_copy();
+  ones_image_sptr->fill(1);
+
+  if (get_compute_with_penalty())
+    objective_function_sptr->accumulate_Hessian_times_input(*kappa_image_target_sptr, *input_image, *ones_image_sptr);
+  else
+    objective_function_sptr->accumulate_Hessian_times_input_without_penalty(*kappa_image_target_sptr, *input_image, *ones_image_sptr);
+
+  // Kappa is defined as the sqrt of the output of accumulate_Hessian_times_input
+  sqrt_image(*kappa_image_target_sptr);
+}
+
+template <typename TargetT>
+void
+KappaComputation<TargetT>::
+compute_kappa_with_approximate()
+{
+  reset_kappa_image_target_sptr(input_image);
+  info("Computing the spatially variant penalty strength using approximate hessian, this may take a while.");
+  // Setup image
+  auto ones_image_sptr = input_image->get_empty_copy();
+  ones_image_sptr->fill(1.);
```

Comment:
Second `fill()` method that is casuing problems with Templates.

Reply:
Instead we can use `this->get_objective_function().fill_nonidentifiable_target_parameters(* ones_image_sptr, 1.);`

Reply:
I had forgotten we have that function. However, I think it does the "complement" (it sets the *non*-identifiable voxels...)

So, another STL one
```c++
std::fill(ones_image_sptr.begin_all(), ones_image_sptr->end_all(), 1.F);
```